rust/lance/src/dataset/scanner.rs

```diff
@@ -1876,6 +1865,12 @@ impl Scanner {
             (**self.dataset.fragments()).clone()
         };
 
+        // If this unwrap fails we have a bug because we shouldn't be using this function unless we've already
+        // checked that there is an index query
+        let index_expr = filter_plan.index_query.as_ref().unwrap();
```

Comment:
might be more of a philosophical question, but why should that crash the server? In a multitenant environment that would bring down all users, not just the one with the messed up install/making the request.

Reply:
FWIW I just ran into this :laughing: 

```
E       ValueError: LanceError(IO): Internal error: compute_projection: expected struct.
E       This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker, /home/pace/dev/lance/rust/lance/src/dataset/scanner.rs:1497:29
```

Fortunately, the string `compute_projection: expected struct` was unique enough to make up for the missing line number.